litmus-portal/backend/graphql-server/graph/schema.resolvers.go

```diff
@@ -5,43 +5,94 @@ package graph
 
 import (
 	"context"
-	"math/rand"
+	"errors"
+	store "github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/data-store"
+	"log"
+	"strconv"
+	"time"
 
 	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/graph/generated"
 	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/graph/model"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/cluster"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database"
 )
 
-var clusterPublish map[string]chan *model.ClusterResponse
+func (r *mutationResolver) UserClusterReg(ctx context.Context, clusterInput model.ClusterInput) (string, error) {
+	return cluster.ClusterRegister(clusterInput)
+}
 
-func init() {
-	clusterPublish = map[string]chan *model.ClusterResponse{}
+func (r *mutationResolver) ClusterConfirm(ctx context.Context, identity model.ClusterIdentity) (string, error) {
+	return cluster.ConfirmClusterRegistration(identity, store.State)
 }
 
-func (r *mutationResolver) CreateCluster(ctx context.Context, input model.ClusterInput) (*model.ClusterResponse, error) {
-	newCluster := &model.ClusterResponse{
-		Data: input.Data,
-		ID:   input.ID,
-	}
+func (r *mutationResolver) NewClusterEvent(ctx context.Context, clusterEvent model.ClusterEventInput) (string, error) {
+	return cluster.NewEvent(clusterEvent, store.State)
+}
 
-	for _, observer := range clusterPublish {
-		observer <- newCluster
+func (r *mutationResolver) NewClusterAction(ctx context.Context, action model.ClusterActionInput) (string, error) {
+	log.Print("NEW CLUSTER ACTION ", action.ClusterID)
+	reqCluster, err := database.GetCluster(action.ClusterID)
+	if err != nil {
+		return "", err
+	}
+	if len(reqCluster) != 1 {
+		return "", errors.New("SERVER ERROR")
 	}
-	return newCluster, nil
+	clusterAction := model.ClusterAction{
+		ProjectID: reqCluster[0].ProjectID,
+		Action:    action.Action,
+	}
+	cluster.SendClusterAction(reqCluster[0].ClusterID, clusterAction, store.State)
```

Comment:
Yes we can, I have kept this here to generalise the function in case we change how the state of the server is managed, we won't have to touch the function code, we can just change the params only.